mypy/newsemanal/semanal_namedtuple.py

```diff
@@ -115,16 +136,21 @@ def check_namedtuple_classdef(
                     default_items[name] = stmt.rvalue
         return items, types, default_items
 
-    def process_namedtuple_definition(self, s: AssignmentStmt, is_func_scope: bool) -> None:
+    def process_namedtuple_definition(self, s: AssignmentStmt, is_func_scope: bool,
+                                      existing_info: Optional[TypeInfo]) -> None:
         """Check if s defines a namedtuple; if yes, store the definition in symbol table.
 
-        Assume that s has already been successfully analyzed.
+        Assume that s has already been successfully analyzed. Use existing_info
+        if given instead of creating a new one.
```

Comment:
As discussed above, I think that it would be better to avoid calling this method if this has been successfully analyzed before.